app/gui/src/presenter/project.rs

```diff
@@ -385,22 +404,26 @@ impl Project {
             eval_ view.execution_context_restart(model.execution_context_restart());
 
             view.set_read_only <+ view.toggle_read_only.map(f_!(model.toggle_read_only()));
+            eval graph_view.execution_environment((mode) model.execution_environment_changed(mode));
+
+            eval_ view.toggle_read_only(model.toggle_read_only());
```

Comment:
Why this line is needed? We already have 

```rust
view.set_read_only <+ view.toggle_read_only.map(f_!(model.toggle_read_only()));
```
 slightly above. I see that read-only mode is broken in a weird way (after toggling it twice removing nodes works, but dragging edges doesn't)